tests/php/json-api/test-class.json-api-plugins-endpoints.php

```diff
@@ -100,6 +100,10 @@ public function test_Jetpack_JSON_API_Plugins_Modify_Endpoint() {
 	 * @requires PHP 5.3.2
 	 */
 	public function test_Jetpack_API_Plugins_Install_Endpoint() {
+		if ( is_multisite() ) {
+			wp_get_current_user()->set_role( 'manage_network');
```

Comment:
This has side effects. The current user will have this role for every subsequent test.

I think we should create a separate user. Maybe something like the following?

```diff
diff --git a/tests/php/json-api/test-class.json-api-plugins-endpoints.php b/tests/php/json-api/test-class.json-api-plugins-endpoints.php
index 3a1f0ac61..4e9803e3d 100644
--- a/tests/php/json-api/test-class.json-api-plugins-endpoints.php
+++ b/tests/php/json-api/test-class.json-api-plugins-endpoints.php
@@ -4,6 +4,12 @@ require_jetpack_file( 'class.json-api.php' );
 require_jetpack_file( 'class.json-api-endpoints.php' );
 
 class WP_Test_Jetpack_Json_Api_Plugins_Endpoints extends WP_UnitTestCase {
+	private static $super_admin_user_id;
+
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$super_admin_user_id = $factory->user->create( array( 'role' => 'administrator' ) );
+		grant_super_admin( self::$super_admin_user_id );
+	}
 
 	public function setUp() {
 		if ( ! defined( 'WPCOM_JSON_API__BASE' ) ) {
@@ -101,7 +107,7 @@ class WP_Test_Jetpack_Json_Api_Plugins_Endpoints extends WP_UnitTestCase {
 	 */
 	public function test_Jetpack_API_Plugins_Install_Endpoint() {
 		if ( is_multisite() ) {
-			wp_get_current_user()->set_role( 'manage_network');
+			wp_set_current_user( self::$super_admin_user_id );
 		}
 
 		$endpoint = new Jetpack_JSON_API_Plugins_Install_Endpoint( array(
```

Reply:
After the test, we'd need to reset the user back to the previous user, correct?

Reply:
Who is "logged in" gets [reset automatically](https://github.com/WordPress/wordpress-develop/blob/5.1.1/tests/phpunit/includes/abstract-testcase.php#L165) (the current user is set to `0`).

~Which… means `wp_get_current_user()->set_role( 'manage_network' )` shouldn't work at all?~

`wp_get_current_user()->set_role( 'manage_network')` does work. It grants the `manage_network` cap to user_id=0 :), and that does not get cleaned up between tests.